ui/addon/src/main/java/org/jboss/windup/ui/RulesetUpdateChecker.java

```diff
@@ -0,0 +1,111 @@
+package org.jboss.windup.ui;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Singleton;
+
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.DependencyResolver;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.forge.addon.dependencies.builder.DependencyQueryBuilder;
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.event.PostStartup;
+import org.jboss.forge.furnace.services.Imported;
+import org.jboss.windup.config.furnace.FurnaceHolder;
+import org.jboss.windup.util.PathUtil;
+
+@Singleton
+public class RulesetUpdateChecker
+{
+
+    private static final String META_INF_VERSION_KEY = "version=";
+    public static final String RULESET_CORE_DIRECTORY = "migration-core";
+
+    public void perform(@Observes PostStartup event)
+    {
+        boolean isDependencies = event.getAddon().getId().toString().contains("org.jboss.windup.ui:windup-ui");
+        if(isDependencies) {
+            Furnace furnace = FurnaceHolder.getFurnace();
+            Imported<DependencyResolver> exportedTypes = furnace.getAddonRegistry().getServices(DependencyResolver.class);
+            if(!exportedTypes.isUnsatisfied() && rulesetNeedUpdate(exportedTypes.get())) {
+                    System.out.println("");
+                    System.out.println("Your ruleset is obsolete. Consider running windup-update-ruleset or windup-update-distribution command. Press ENTER to continue.");
+                    System.out.println("");
+            }
+            
+        }
+    }
+
+    public static boolean rulesetNeedUpdate(DependencyResolver dependencyResolver)
+    {
+        List<Coordinate> resolveVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder.create(CoordinateBuilder.create()
+                    .setGroupId("org.jboss.windup.rules")
+                    .setArtifactId("windup-rulesets")));
+       
+        int i = 0;
+        Coordinate latestCoordinate;
+        do{
```

Comment:
It looks like this file is formatted incorrectly. Could you run the formatter please and rebase this?